opcua/client.py

```diff
@@ -357,7 +357,11 @@ def activate_session(self, username=None, password=None, certificate=None):
             params.UserIdentityToken.UserName = username 
             if self.server_url.password:
                 pubkey = uacrypto.x509_from_der(self.security_policy.server_certificate).public_key()
-                data = uacrypto.encrypt_basic256(pubkey, bytes(password, "utf8"))
+                # see specs part 4, 7.36.3: if the token is encrypted, password
+                # shall be converted to UTF-8 and serialized with server nonce
+                etoken = ua.pack_bytes(bytes(password, "utf8") + self._server_nonce)
+                #data = uacrypto.encrypt_basic256(pubkey, etoken)
```

Comment:
encrypt_basic256 was not tested. so you are probably correct. I will merge and do a few tests when I get two minutes